test/distributed/org/apache/cassandra/distributed/test/ReadRepairTest.java

```diff
@@ -136,55 +214,129 @@ public void movingTokenReadRepairTest() throws Throwable
+            }
 
-            consume(cluster.coordinator(2).executeWithPaging("SELECT * FROM distributed_test_keyspace.tbl WHERE key = ? and column1 >= ? and column1 <= ?",
-                                           ConsistencyLevel.ALL, 1,
-                                           "test", 8, 12));
+            // delete a subset of the inserted partitions, plus some others that don't exist
+            String deletePartitionQuery = withKeyspace("DELETE FROM %s.t WHERE k = ?");
+            for (int k = 5; k < 15; k++)
+            {
+                coordinator.execute(deletePartitionQuery, ALL, k);
+            }
 
-            consume(cluster.coordinator(2).executeWithPaging("SELECT * FROM distributed_test_keyspace.tbl WHERE key = ? and column1 >= ? and column1 <= ?",
-                                                             ConsistencyLevel.ALL, 1,
-                                                             "test", 16, 20));
-            cluster.get(2).executeInternal("DELETE FROM distributed_test_keyspace.tbl WHERE key=? AND column1>? AND column1<?;",
-                                           "test", Integer.MIN_VALUE, Integer.MAX_VALUE);
+            // delete a some of the rows of some of the partitions, including deleted and not deleted partitions
+            String deleteRowQuery = withKeyspace("DELETE FROM %s.t WHERE k = ? AND c = ?");
+            for (int k = 2; k < 7; k++)
+            {
+                for (int c = 0; c < 5; c++)
+                    coordinator.execute(deleteRowQuery, ALL, k, c);
+            }
+
+            // delete a some of the rows of some not-existent partitions, including deleted and never-written partitions
+            for (int k = 12; k < 17; k++)
+            {
+                for (int c = 0; c < 5; c++)
+                    coordinator.execute(deleteRowQuery, ALL, k, c);
+            }
+
+            // flush all the nodes if specified
+            if (flush)
+            {
+                for (int n = 1; n <= cluster.size(); n++)
+                    cluster.get(n).flush(KEYSPACE);
+            }
+
+            // run an unrestricted range query with tracing
+            final UUID sessionId = UUIDGen.getTimeUUID();
+            coordinator.executeWithTracing(sessionId, withKeyspace("SELECT * FROM %s.t LIMIT 100"), QUORUM);
+            List<TracingUtil.TraceEntry> traces = TracingUtil.getTrace(cluster, sessionId);
+
+            // use the trace to verify that the previous query hasn't triggered any read-repair mutation
+            traces.forEach(traceEntry -> {
+                String activity = traceEntry.activity;
+                String errorMsg = "Found unexpected read-repair mutations";
+                assertFalse(errorMsg, activity.contains("Appending to commitlog"));
+                assertFalse(errorMsg, activity.contains("Adding to t memtable"));
+                assertFalse(errorMsg, activity.contains("Acquiring switchLock read lock"));
+            });
```

Comment:
@adelapena What if we checked the read-repair metrics in `TableMetrics`, sort of like what we do in `ReplicaFilteringProtectionTest`? It might be a little less brittle than string matching?

Reply:
nit: It might also be interesting to issue a couple restricted queries against the specific cut points from the earlier series of operations to verify that they don't actuate any RR. (This might not even be that bloated if we're just checking the table metrics and can avoid tracing.)

Reply:
I have added more queries and replaced the tracing verifications by a much simpler metrics check.